[Merged by Bors] - 274 document the bundlebuilder and rego rule configmaps (#350)
stackabletech/opa-operator: docs/modules/ROOT/pages/implementation-notes.adoc

== OPA Bundle Builder

Users can manage the Rego-rules evaluated by an OPA cluster by creating, updating and deleting `ConfigMap` resources.

> The role of the bundle builder is to convert these ressources to bundles (`tar.gz` files) and make it available as a HTTP endpoint.

Reviewer: ```suggestion
The role of the bundle builder is to convert these resources to bundles (`tar.gz` files) and make them available via an HTTP endpoint.
```

Reviewer: The "ressources" is not fixed yet?

Author: I made the same changes locally that's why I couldn't apply your suggestion. It's fixed now.

> The bundle builder runs in a side container of the OPA `Pod` as a simple HTTP server that OPA querying regularly

Reviewer: ```suggestion
The bundle builder runs in a side container of the OPA `Pod` as a simple HTTP server that OPA is querying regularly
```

(every 20 to 30 seconds) for updates.

NOTE: Kubernetes limits the size of `ConfigMap`s to 1MB. Users have to take this limit into consideration when manging Rego-rules.

Only `ConfigMap`s labeled with `opa.stackable.tech/bundle` are watched by the builder when updating bundles. The name of
the `data` entries in the `ConfigMap` are used as file names when storing the rules in the bundle.

> NOTE: Currently it is the user's responsability to make sure these names do not collide.

Reviewer: ```suggestion
NOTE: Currently it is the user's responsibility to make sure these names do not collide (as they will override each other).
```
